tiledb/sm/query/query.cc

```diff
@@ -859,8 +860,13 @@ Status Query::process() {
           // Make sure all ranges are contained in the current domain.
           for (auto& range : subarray_.ranges_for_dim(d)) {
             if (!cd->includes(d, range)) {
-              throw QueryException(
-                  "A range was set outside of the current domain.");
+              throw QueryException(std::format(
+                  "A range {} on dimension '{}' was set outside of the current "
+                  "domain {}.",
+                  range_str(
+                      range, array_schema_->domain().dimension_ptr(d)->type()),
+                  array_schema_->domain().dimension_ptr(d)->name(),
```

Comment:
Shouldn't this be `current_domain()`?

Reply:
We need the domain to get types. Those are wired through `array_schema_` and `current_domain` both. I can get them either way.

Reply:
@ihnorton correction: the domain is wired through the `array_schema_` and the private `ndrectangle_` of `CurrentDomain`.

It is an invariant that they be the same domain in both places.

If you want to pursue the latter then we need to connect a domain accessor in `CurrentDomain` which passes through the copy of the domain in its `ndrectangle_`. Personally I don't see the benefit to doing the latter.